Fix CreateVolumeResponse after snapshot restore

jsafrane opened the pull request:
CSI wants CreateVolumeResponse.ContentSource to be set to the snapshot that has just been restored.

Fixes #444

jsafrane commented:
/assign @wongma7

bertinatto commented:
/lgtm

leakingtapan commented on pkg/driver/controller.go:
```diff
@@ -154,9 +154,14 @@ func (d *controllerService) CreateVolume(ctx context.Context, req *csi.CreateVol
 
+	snapshotID := ""
+	if snapshotSource := req.VolumeContentSource.GetSnapshot(); snapshotSource != nil {
+		snapshotID = snapshotSource.SnapshotId
```
Do we need to set the snapshot ID for an existing volume? Or only at line 188 here 
https://github.com/kubernetes-sigs/aws-ebs-csi-driver/pull/446/files#diff-61566437d9b17d18936960aa0bce1ad4R187-R196

for restoring/creating the volume out of snapshot

jsafrane replied:
Yes, for idempotency. Responses to repeated CreateVolume requests for snapshot restore must have the same fields as the initial response.

leakingtapan commented on pkg/driver/controller.go:
```diff
@@ -154,9 +154,14 @@ func (d *controllerService) CreateVolume(ctx context.Context, req *csi.CreateVol
 	// volume exists already
 	if disk != nil {
-		return newCreateVolumeResponse(disk), nil
+		return newCreateVolumeResponse(disk, snapshotID), nil
```
I would prefer snapshot ID be part of the disk struct

jsafrane replied:
Added. But it moves us further away from https://github.com/kubernetes-sigs/aws-ebs-csi-driver/issues/393

leakingtapan commented:
Could you add some unit test for it? thx

jsafrane commented:
> Could you add some unit test for it?

There were no snapshot restore tests. I added some.

I added also a check if existing volume was created from expected snapshot, for repeated requests.

jsafrane commented:
`VpcLimitExceeded: The maximum number of VPCs has been reached.`
/retest

leakingtapan commented:
/lgtm
/approve
